### PR TITLE
Extensions to see if a member/role "canInteract" with another Member/Role

### PR DESCRIPTION
**Pull request by ByteAlex**
Pretty much self explaining I guess

### CODE REVIEW

File: kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/utils/_Member.kt [hunk 1]
```diff
@@ -85,3 +86,36 @@ public suspend fun Member.hasPermissions(perms: Collection<Permission>): Boolean
...
+ * The logic also accounts for [Guild] ownership.
+ *
+ * Throws an [IllegalArgumentException] if the role is from a different guild.
+ */
+public suspend fun Member.canInteract(role: Role): Boolean {
+    val guild = getGuild()
+    if (guild.ownerId == this.id) return true
+    val highestRole = getTopRole() ?: guild.getEveryoneRole()
+    return highestRole.canInteract(role)
+}
```
Comment (gdude2002) [inline kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/utils/_Member.kt:105]: This needs way more blank lines.

File: kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/utils/_Member.kt [hunk 2]
```diff
@@ -85,3 +86,36 @@ public suspend fun Member.hasPermissions(perms: Collection<Permission>): Boolean
...
+ *
+ * Throws an [IllegalArgumentException] if the member is from a different guild.
+ */
+public suspend fun Member.canInteract(member: Member): Boolean {
+    val guild = getGuild()
+    if (isOwner()) return true
+    if (member.isOwner()) return false
+    val highestRole = getTopRole() ?: guild.getEveryoneRole()
+    val otherHighestRole = member.getTopRole() ?: guild.getEveryoneRole()
+    return highestRole.canInteract(otherHighestRole)
```
Comment (gdude2002) [inline kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/utils/_Member.kt:125]: This needs waaaaaay more blank lines.

File: kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/utils/_Member.kt [hunk 3]
```diff
@@ -85,3 +86,36 @@ public suspend fun Member.hasPermissions(perms: Collection<Permission>): Boolean
...
+}
```
Comment (gdude2002) [inline kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/utils/_Member.kt]: Don't remove the newlines at the end of the files. How have you been using Git this long without having IJ add the newline for you?

File: kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/utils/_Role.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+/**
+ * Checks whether a [Role] can interact with another [Role] by comparing their [rawPosition]s.
+ *
+ * Throws an [IllegalArgumentException] when the roles are not from the same guild.
+ */
+public fun Role.canInteract(role: Role): Boolean {
+    if (role.guildId != guildId)
+        throw IllegalArgumentException("canInteract can only be called within the same guild!")
+    return role.rawPosition < rawPosition
+}
```
Comment (gdude2002) [inline kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/utils/_Role.kt]: Newlines. Both in the function, and at the end of the file. Also, this conditional is on more than one line, so it needs braces.

**Review by gdude2002 — CHANGES_REQUESTED** [review]

File: kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/utils/_Member.kt (hunk 3 above)
Comment (ByteAlex) [reply kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/utils/_Member.kt]: Never did, IJ never complained. Small files!

**Comment by gdude2002** [comment]
:rocket:

**Pull request merged**